### ISSUE TITLE
Fusion Execution Error with Nested Query

### ISSUE THREAD

**Issue by ndejaco2** [ChilliCream/graphql-platform#6497]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Hot Chocolate

### Describe the bug

I have the following two schemas:

```
type Author {
	id: ID!
	name: String!
	bio: String
	contactEmail: String
	nationality: String
}

type AuthorConnection {
	items: [Author]
	nextToken: String
}

type Book {
	id: ID!
	authorId: ID!
	author: Author
}

input CreateAuthorInput {
	name: String!
	bio: String
	contactEmail: String
	nationality: String
}

input DeleteAuthorInput {
	id: ID!
}

type Mutation {
	createAuthor(input: CreateAuthorInput!): Author
	deleteAuthor(input: DeleteAuthorInput!): Author
}

type Query {
	authorById(id: ID!): Author
	authors(limit: Int, nextToken: String): AuthorConnection
	bookByAuthorId(authorId: ID!): Book
}

schema {
	query: Query
	mutation: Mutation
}
```

```
type Author {
	id: ID!
	books: BookConnection
}

type Book {
	id: ID!
	title: String!
	authorId: ID!
	publisherId: ID
	genre: String
	publicationYear: Int
}

type BookConnection {
	items: [Book]
	nextToken: String
}

input CreateBookInput {
	title: String!
	authorId: ID!
	publisherId: ID
	genre: String
	publicationYear: Int
}

input DeleteBookInput {
	id: ID!
}

type Mutation {
	createBook(input: CreateBookInput!): Book
	deleteBook(input: DeleteBookInput!): Book
}

type Query {
	bookById(id: ID!): Book
	books(limit: Int, nextToken: String): BookConnection
	authorById(id: ID!): Author
}

schema {
	query: Query
	mutation: Mutation
}
```

I cant successfully execute a nested query with multiple joins like so:

```
query Query {
  authorById(id: "d66a6eec-3571-4df9-8cd4-693cd5aa99e3") {
    id,
    name,
    bio,
    nationality,
    books {
      items {
        id
        author {
          id,
          books {
            items {
              id
            }
          }
        }
      }
    }
  }
}

```
It appears that fusion runs into an issue with the final books resolver. It will omit that field in the response. If I remove the id field at the same level the query returns a 500. When run under debug, I get the following error message:

```
{
  "errors": [
    {
      "message": "Unexpected Execution Error",
      "extensions": {
        "message": "A selection set must not be empty.",
        "stackTrace": "   at HotChocolate.Fusion.Planning.RequestDocumentFormatter.CreateSelectionNodes(QueryPlanContext context, SelectionExecutionStep executionStep, IObjectType possibleType, SelectionSet selectionSet, List`1 selectionNodes)\n   at HotChocolate.Fusion.Planning.RequestDocumentFormatter.CreateSelectionSetNode(QueryPlanContext context, SelectionExecutionStep executionStep, ISelection parentSelection)\n   at HotChocolate.Fusion.Planning.RequestDocumentFormatter.CreateSelectionNode(QueryPlanContext context, SelectionExecutionStep executionStep, ISelection selection, ObjectFieldInfo fieldInfo)\n   at HotChocolate.Fusion.Planning.RequestDocumentFormatter.CreateRootSelectionSetNode(QueryPlanContext context, SelectionExecutionStep executionStep)\n   at HotChocolate.Fusion.Planning.RequestDocumentFormatter.CreateRequestDocument(QueryPlanContext context, SelectionExecutionStep executionStep, OperationType operationType)\n   at HotChocolate.Fusion.Planning.Pipeline.ExecutionNodeBuilderMiddleware.CreateResolveNode(QueryPlanContext context, SelectionExecutionStep executionStep)\n   at HotChocolate.Fusion.Planning.Pipeline.ExecutionNodeBuilderMiddleware.Invoke(QueryPlanContext context, QueryPlanDelegate next)\n   at HotChocolate.Fusion.Planning.QueryPlanPipelineBuilder.<>c__DisplayClass5_1`1.<Use>b__1(QueryPlanContext context)\n   at HotChocolate.Fusion.Planning.Pipeline.RequirementsPlannerMiddleware.Invoke(QueryPlanContext context, QueryPlanDelegate next)\n   at HotChocolate.Fusion.Planning.QueryPlanPipelineBuilder.<>c__DisplayClass4_0`1.<Use>b__1(QueryPlanContext context)\n   at HotChocolate.Fusion.Planning.Pipeline.FieldRequirementsPlannerMiddleware.Invoke(QueryPlanContext context, QueryPlanDelegate next)\n   at HotChocolate.Fusion.Planning.QueryPlanPipelineBuilder.<>c__DisplayClass5_1`1.<Use>b__1(QueryPlanContext context)\n   at HotChocolate.Fusion.Planning.Pipeline.ExecutionStepDiscoveryMiddleware.Invoke(QueryPlanContext context, QueryPlanDelegate next)\n   at HotChocolate.Fusion.Planning.QueryPlanPipelineBuilder.<>c__DisplayClass5_1`1.<Use>b__1(QueryPlanContext context)\n   at HotChocolate.Fusion.Planning.QueryPlanner.Plan(IOperation operation)\n   at HotChocolate.Fusion.Execution.Pipeline.OperationQueryPlanCompiler.OptimizeOperation(OperationOptimizerContext context)\n   at HotChocolate.Execution.Processing.OperationCompiler.CreateOperation(String operationId, OperationDefinitionNode operationDefinition, ObjectType operationType, DocumentNode document, ISchema schema)\n   at HotChocolate.Execution.Processing.OperationCompiler.Compile(String operationId, OperationDefinitionNode operationDefinition, ObjectType operationType, DocumentNode document, ISchema schema, IReadOnlyList`1 optimizers)\n   at HotChocolate.Execution.Pipeline.OperationResolverMiddleware.CompileOperation(IRequestContext context, String operationId, OperationDefinitionNode operationDefinition, ObjectType operationType)\n   at HotChocolate.Execution.Pipeline.OperationResolverMiddleware.InvokeAsync(IRequestContext context)\n   at HotChocolate.Execution.Pipeline.OperationComplexityMiddleware.InvokeAsync(IRequestContext context)\n   at HotChocolate.Execution.Pipeline.OperationCacheMiddleware.InvokeAsync(IRequestContext context)\n   at HotChocolate.Execution.Pipeline.DocumentValidationMiddleware.InvokeAsync(IRequestContext context)\n   at HotChocolate.Execution.Pipeline.DocumentParserMiddleware.InvokeAsync(IRequestContext context)\n   at HotChocolate.Execution.Pipeline.DocumentCacheMiddleware.InvokeAsync(IRequestContext context)\n   at HotChocolate.Execution.Pipeline.TimeoutMiddleware.InvokeAsync(IRequestContext context)\n   at HotChocolate.Execution.Pipeline.ExceptionMiddleware.InvokeAsync(IRequestContext context)"
      }
    }
  ]
}
```

From my understanding I have correctly declared the schema as the first books resolver is able to correctly join the author data with books data.

### Steps to reproduce

1. Create apis with the schema that I shared above.
2. Run the query that I shared above. 

### Version

13.6.0-preview.3

**Comment by michaelstaib** [comment]
Thanks for reporting... we found this already internally and are fixing the query planner at the moment.

**Comment by michaelstaib** [comment]
If you are interested we can add you to our biweekly fusion discussions with the guild.

**Comment by ndejaco2** [comment]
That would be great. I have a failing test created if that is useful and am happy to dig more if that is of value.

**Comment by michaelstaib** [comment]
Can you create a pr for the test?

**Comment by michaelstaib** [comment]
If you post me your email to contact@chillicream.com I can bring you into the company slack.

**Comment by ndejaco2** [comment]
Yea sure will do now. My email is ndejaco123@gmail.com. Thanks!

**Comment by ndejaco2** [comment]
Using my failing test I figured out the bug. It has to do with how the exports between steps are being handled. The RequirementsPlannerMiddleware is used a map of subgraph name to providing execution step which is used when determining which step is a provider of a given export. The problem is that when you have multiple calls to the same subgraph in the same query that needs to provide exports, it currently only considers the first step as the one that would be able to provide exports (TryAdd fails when registering a second call to the same subgraph but return value is not checked) . A subsequent step that needs an export is not able to look up the registered export it needs because the logic is keyed on the providing execution step.

**Comment by ndejaco2** [comment]
I think switching this line:
schemas.TryAdd(siblingExecutionStep.SubgraphName, siblingExecutionStep) 

To:
schemas[siblingExecutionStep.SubgraphName] = siblingExecutionStep;

Fixes the issue as it ensures that we have exports provided by the most recent call to a given subgraph. This causes some tests to fail as the snapshots do not match. Is there any reason where getting the export from the most recent call would not be desired?

**Comment by ndejaco2** [comment]
I went ahead and added that potential fix to my PR

**Comment by ndejaco2** [comment]
Fixed with #6504
